### ISSUE TITLE
Fix false positives for parens in selectors in indentation

### ISSUE THREAD

**Issue by fregante** [stylelint/stylelint#5181]
Initially reported in https://github.com/stylelint/stylelint/issues/3461#issuecomment-788214666

> Clearly describe the bug

`indentation` does not cover the indentation of the content of parens in selectors, like `:is()`

> What code is needed to reproduce the bug?

```css
:is(
a,
b
) {}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "indentation": 2
  }
}
```

> Which version of stylelint are you using?

13.11.0

> What did you expect to happen?

`a` and `b` should be indented:
```css
:is(
  a,
  b
) {}
```

> What actually happened (e.g. what warnings or errors did you get)?

They're accepted as is, without indentation.

> Context

_Originally posted by @jeddy3 in https://github.com/stylelint/stylelint/issues/3461#issuecomment-788765380_

> We [apply indentation to parens in at-rules and declaration-values](https://stylelint.io/user-guide/rules/indentation#indentinsideparens-twiceonce-at-root-twice-in-block), but not those in selectors. It should, though.
>

**Comment by jeddy3** [comment]
@fregante Thanks for the report and for using the template.

I can reproduce using [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcCCWBnAFAHQHYAICGANNjgEbYCUOwxOA7klAC4AW8NOOYeANmJrh0EBGAAwiApDgC0OUQAcAHuxzkA3NgC+2EARAoAnlkZ4FCcChQ7wELADMkAczPUBGEACcArtxgo38KmU3PEYpLx8pLAgpTywAayjaLH8cRi8YIgEONxJuCDA4yOiYAFs5Rn0UtM8MoOs89yKpJCwANx4GKWYYJQRU9MzBHDdIEpKYIybS8sq+6tqs4ZBYMG48dxCkGylc-MKoqShPOW4kLkYYKTl3CDkYd0YkXxSAbWUOecGhwMWht0cou4YK93t8RjYUDAwJ5Hq1LkcTmcQr4pPQWIckLZbPcJqF2twan4QKCOABdUFaRakr7ZZZQtYbR7bXYFJooZgQB7MPBYKBXG53B76KQQOHudwMZ5zAZ1Ww2UK2PAlJDcYUHBGnc6XLBKqUBT6y+VSRXK1VNZUWFoOKQOCb3M7GpUq4VxGD6WicqBVGWLNy2WJgJlYKRcXhNFr4hjemo0pb+rCBrbB0NgJqxFByPBgGB8273EKc6MLP4geOJ7anLAwdY2jZQJ6TA5RLAoYy89Z8+tA8vJaUxuqu-S2DbjQ709abbYHJBlTltxhF2Nucb1vDG6uMTxAyK6tPxRK9-U+kvq3zZ3lWqQZqGYs5ICqLurq46a5FSELNWcPMLePX9fu+iAz6IlqV4wD4gackSR4ASexRlBUV4QFu2aPoBBw9Gka6QsqkB5C2aFwc0bQdJ2yG5JcKBrOyIYQGMuLQf+xaCG41x5kKe4JBASSESxejgVCjCclckKeFA0SrHgFicQevG0pCEFCY0ciieJUgCeMjbRLEXE8X2zHyQJkGNBUdwydxh5MUuehpJeTYwLQlbAvp1mxPe5l6TBBlLB+4SXNMiFOTsMBykCIK-D53C0Hg+hErGHAuN8fE9Nm5ThUlJYsnEPjSXgtgXI0KC7jqo5ZTlcUknx9juK2kS+BcXrEhFZLxXxAKcs5ARvM1fF5QVtH0UYbiVeSEWUkM1J1L5v47qOXCQikbh5LQ9xuNZ00RCVlGZtm775atfTBFFMUoFeVoRE5a1TaEflgbhED4XVjktJcfUHQER3RbFV2AVlIZ5Cgl4kBsu0BcKQUkCFHWLUBMBij9mV5KyqwQIDWDWsDWbag5QVve4MM8F9FW-UjhQo2jGMg9jz1VsFoWdZFRNSCU3iPFIl06HUf3k0DVNXjtlyQ-TBPHbFZ3oxdL0I3xf15lgvNY09uP7fjh0gITJ3M6zSDs1LnMk3sIp3PL6M7HzGaK3jItMxTktVtLtKy8bCu7Rbu1C9Dasa99+slnhwndAoIZSQzS3cQd1n+40gfsxMDgsDD7Jzg7SyjJpoRg7rtMe2FfTdUln0nSnCWVW4KUwGleeVbS1W1VWrY5sNPWTT1-wOICDP5xl8kVD4lahGnPJQESI0UsoLd+3R6eosw96+ALxGA7A1tF77fHQq2dH8uxiGZxDUO511oKFz7rWJRlZcKKlC5Vz1tJ4yGnibyU2+Cg+a-d36SA1aE9cNU33cJ4XxAO1Q+OAu6f3VirAa6di5ghAS2SU4sHB20FqTABGVRpJXGoIIBtIVgMknMGEgPJrRu1eirGGVZ4YfyWAQicQYdikP5orHOod1ai2JiWehjIkw7FJndM4D0pw4xentAqK8xYs24GzDm1keFEP4YbHCQjHrkPEe9RmmtbaXDkXUBRjC-oqP9sGdRbCqFwwjvo8cvDmQCJ0VnS4JQTBjkkrYgifQhDyJsYov6WEVSXmMcIyyJ8uF8QMXwkxSsxFWy9pwrWMidZ6MAhE7YUT1GxI+hwm251dF628W4xR6SF7mLVtQqxKSfGML3mI0pR8IqhLgefAuIBy6V3qd3e+0DUm9laiWWuv96qNyaoAvpSxQGd2rj5aBadcRwJLC0JelFcmOKUQUDBSUsHfBwS1WUAZGGD2ibTTJWipHa0cSnP0+zIlT2wgvE5oTkGoMuaWa5aTbksPdgfdh5TVbWTLIw5xQcan1xSCIf5bzgxbWDgtNWy0Kn9MhVcdYuJuiQlOlWGmlxFmSkkadaRsj8l7ITIwzMQIjBopROonFy8ymWMKis5JiKSV8NoLPC4GTKFxKJi8losAjBEJSAAJmsj4dG6IABe9xogHDcjfLy1kgXqQQuDF6jEvF1BXEgNc2IQhbkuMUy2XKsnezCbSLVOqNz6tomokp3yLE0MVTmbV649XbmhfNdh8K-maudZat1lwyWot8FSheNKfn0peRa11m5twbHRpcdiBZCr3ONacs1Sxo26tjZceNtojb5iUp8wW9ruWrydauKQABHGo7hVW1UOZi5WEiy1nMSRc2hy4-XVtrfWgeHzOUtpNfEhxTK+LRprfcPtMC7msNLVk35LyMIPRnnPchcklgHAtBTdSvIRS2CQihamHMFVPk8CUSGjQfB4HrKbKVNw8VLvPZepo-jKzWnvajDdrEBT3EQp6mG3qXm3RBSW4Wt8WmmoSYS+2tCS6t1aVfCu8rwFTK-j-OqDdGojTGW3Du6VOnBBmVPOZIzMFjypNZBSglhIhFsiQGEgsQaukYKdaliDaULsjZ2-iilaOMHo4xgtjJhKDs0W4RdPHqPGXfAJiUDGLjCeTcWumnsuOOrqNJotowSAtGU2J1Ww6eVSaMtpuiumdRFrMfO9NLytPCVgCgc8UAeT9ovXpotTZtjrpcpp0zwlTgNo+U2mJabGkmb41e1AbnnEqbqZu7jVH-ONCVaBxi4K-ORZEjAMSElqKnQA3C8OPrAL2eUqpPLUlTpBopSGtjYaOMRo06V5L2XcvqR8NPQrWSgMRZo+VnLakNK4htVOCAxggww3Ep4CidnWumQNSHQDxWXlyphV6lbPH8Q1CkIOD07g+TdaWL16y228lBbGGuELxywvZM1gSpJRLAJnd1hd2LBm8VPLybB07PAduBZi7Or54H1MIr4i9gHzMXFpbBTQSkGgQAaCAA) with:

```css
:is(
 a,
 b
) {
  width:
    calc(
      100% - 10px
    );
}
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by fregante** [comment]
Interestingly enough, `:is(:is())` is partially indented:

```css
:is(
:is(
  a,
  b
),
b
) {
  width:
    calc(
      100% - 10px
    );
}
```

Whereas `:is(:is(:is()))` just fails between `selector-pseudo-class-parentheses-space-inside` and `indentation`

```css
:is(
:is(
  :is(
    a,
    b
  ) /* Unexpected whitespace before ")" */
)
) {
  width:
    calc(
      100% - 10px
    );
}
```

**Comment by jeddy3** [comment]
Thanks for digging deeper.

> Whereas `:is(:is(:is()))` just fails between `selector-pseudo-class-parentheses-space-inside` and `indentation`

It looks like we'll need to bring the same flexibility we have for functions to pseudo-classes, i.e. add a:
- `never-single-line` option to `selector-pseudo-class-parentheses-space-inside`, like [function](https://stylelint.io/user-guide/rules/function-parentheses-space-inside#never-single-line)
- `selector-pseudo-class-parentheses-newline-inside` rule, like [function](https://stylelint.io/user-guide/rules/function-parentheses-newline-inside)

If anyone fancies tackling these, please create new issues for each so we can track them.

---

Incidentally, it's amazing to see how much `:is()` cleaned things up in that referenced pull request.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.

**Comment by fregante** [comment]
I think basic indentation should be a core part of a linter — and it is. Stylelinter already does this, but this issue suggests that it's not done correctly/fully. This is a bug in an existing feature, not a new feature.
